lib/logstash/inputs/gelf.rb

```diff
@@ -96,7 +96,21 @@ def udp_listener(output_queue)
       # Gelfd parser outputs null if it received and cached a non-final chunk
       next if data.nil?
 
-      event = LogStash::Event.new(LogStash::Json.load(data))
+      begin
+        event = LogStash::Event.new(LogStash::Json.load(data))
+      rescue LogStash::Json::ParserError => e
+        @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
+        event = LogStash::Event.new("message" => data, "tags" => ["_jsonparsefailure"])
+      rescue StandardError => e
```

Comment:
As long there's some reassurance that https://github.com/logstash-plugins/logstash-input-gelf/issues/26 won't happen if I follow your suggested approach, I'll be happy to have a look at how `json_lines` does it.

I don't care much for rescuing `StandardError` either, but an unreliable logging mechanism seems like a much bigger problem from the user's perspective. :)